pkg/api/types.go

```diff
@@ -1753,6 +1753,36 @@ type ReplicationControllerStatus struct {
 
 	// ObservedGeneration is the most recent generation observed by the controller.
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// Represents the latest available observations of a replication controller's current state.
+	Conditions []ReplicationControllerCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+type ReplicationControllerConditionType string
+
+// These are valid conditions of a replication controller.
+const (
+	// ReplicationControllerReplicaFailure is added in a replication controller when one of its pods
+	// fails to be created or deleted.
+	ReplicationControllerReplicaFailure ReplicationControllerConditionType = "ReplicaFailure"
+	// ReplicationControllerReady denotes whether all of the replicas for the controller are ready or not.
```

Comment:
Yes, this will flip. Actually you got me thinking that this information already exists in the resource printer of a replica set (READY pods). I wanted to make replica sets more usable but this condition doesn't add much. Removing...